Reflection/Interfaces.simba

```diff
@@ -0,0 +1,199 @@
+{$include_once Internal/Reflection.simba}
+{$include_once Constants.simba}
+{$include_once Menu.simba}
+{$include_once Inventory.simba}
+{$include_once Mouse.simba}
+{$include_once Timing.simba}
+
+Function R_WidgetIsHidden(Constant: RWidget): Boolean;
+var
+  Widget: RSWidget;
+begin
+  Widget := RSWidget.Get(Constant.Group, Constant.Child);
+  if Widget.ref = nil then
+    Exit(False);
+
+  Result := Widget.IsHidden;
+  Widget.Free;
+end;
+
+(*
+MAGIC SPELL BOOK
+~~~~~~~~~~~~~~~~
+
+Valid spells are located in Constants.simba
+  Ex: R_SPELL_VARROCK_TELEPORT
+
+*)
+
+Function R_SpellBook: Boolean;
+begin
+  Result := RSWidget.IsValid(R_SPELLBOOK_CONTAINER.Group, R_SPELLBOOK_CONTAINER.Child);
+end;
+
+Function R_WaitSpellBook(timeout: UInt32): Boolean;
+begin
+  Result := R_WaitFunction(timeout, 50, @R_SpellBook);
```

Comment:
In the latest Simba, there is a function called `WaitUntil`. It is used like (for example):

```pascal
WaitUntil(R_SpellBook, interval, timeout);
WaitUntil(SomeFunc(....) = result, interval, timeout);

WaitUntil(R_CountInventoryItem(item) > 14, 50, 1000);
```

This means, you no longer have to do `R_WaitFunction`

Reply:
Oh! Okay! I just noticed you used this in the Bank functions and didn't want to stray from what you were doing initially. That would need to be updated as well.